core/src/main/java/com/linecorp/armeria/server/RouteBuilder.java

```diff
@@ -543,32 +541,12 @@ private static PathMapping getPathMapping(String pathPattern) {
                     "pathPattern: " + pathPattern +
                     " (not an absolute path starting with '/' or a unknown pattern type)");
         }
-        if (!pathPattern.contains("{") && !pathPattern.contains(":")) {
+        if (!pathPattern.contains("/{") && !pathPattern.contains("/:")) {
```

Comment:
> { is not an allowed character so it is percent-encoded by Armeria client or server. It means a request whose path is /path/infix{foo} fails to match the service for /path/infix{foo} and returns 404.

I agree with this sentiment. However, I also thought that not '{}' but also other characters are valid path patterns but have the same issue. e.g. `/$@` will have a similar issue.
Because of this, I thought 1) this was easy enough for maintainers to reason about 2) users won't be using this pattern anyways.
Let me know if you think strict validation is necessary though. Perhaps we can consider it as a separate issue.

Reply:
`/path/infix{foo} is not a common pattern for REST API. Let's resolve the problem when we supporting the use of a path variable with literal characters like https://github.com/spring-projects/spring-framework/blob/main/spring-core/src/test/java/org/springframework/util/AntPathMatcherTests.java#L659-L661